From 8a499f5ae7fc293653c111debb54e634e632f0e7 Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Mon, 22 Apr 2013 18:32:58 +0300
Subject: [PORT FROM MAIN] Telephony: Fix Locale construction when language code is null

BZ: 34594

When the language code is null, instead of constructing
with empty language code and valid country code, Locale
is constructed with country code provided as a Language
Code.

Framework calls the Locale with country code as language
code, this patch fixes this issue.

Change-Id: I83382f3c814d9fb3c762807eba86b440ef9d43b9
Orig-Change-Id: Ib8fe45b2aa58079ce0463b63205111eb17b95664
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../com/android/internal/telephony/MccTable.java   |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/java/com/android/internal/telephony/MccTable.java b/src/java/com/android/internal/telephony/MccTable.java
index 0839f2d..43a8a6b 100644
--- a/src/java/com/android/internal/telephony/MccTable.java
+++ b/src/java/com/android/internal/telephony/MccTable.java
@@ -102,7 +102,7 @@ public final class MccTable
         } else {
             Locale locale;
             if (entry.mLanguage == null) {
-                locale = new Locale(entry.mIso);
+                locale = new Locale("", entry.mIso);
             } else {
                 locale = new Locale(entry.mLanguage, entry.mIso);
             }
-- 
1.7.4.1

